components/fonts/platform/freetype/font.rs

```diff
@@ -128,23 +131,33 @@ impl PlatformFontMethods for PlatformFont {
             return Err("Could not memory map");
         };
 
-        // Variable fonts, where the font designer has provided one or more axes of
-        // variation do not count as font synthesis and their use is not affected by
+        // If the font face is already bold, applying synthetic bold would "double embolden" the font
+        let table_provider_data = FreeTypeFaceTableProviderData::Local(
+            Arc::new(memory_mapped_font_data),
+            font_identifier.index(),
+        );
+        let face_is_bold = table_provider_data
+            .font_ref()
+            .and_then(|font_ref| font_ref.os2())
```

Comment:
This might be able to be a little simpler:

```suggestion
            .and_then(FontRef::os2)
```

Reply:
This actually doesn't work because `and_then` would expect a closure `FnOnce(FontRef<'_>) -> TableRef<'_>` but `FontRef::os2` takes a reference (ie. `FnOnce(&FontRef<'_>) -> TableRef<'_>`)